src/app/user/user.service.ts

```diff
@@ -1,8 +1,65 @@
+import { Response, Request } from 'express';
+import { FilterQuery } from 'mongoose';
+import { IUser } from '../user/user.type';
+
 import User from './user.model';
+
 export class UserService {
+
   private model = User;
 
-  getUser(): string {
-    return '';
+  // get User
+  async getUser(req: Request, res: Response, filter: FilterQuery<IUser>): Promise<void> {
+    try {
+      const user: IUser | null = await this.model.findOne(filter);
+      res.status(200).json({ user })
+    } catch (error) {
+      throw error
+    }
+  }
+
+  // add User
+  async addUser(req: Request, res: Response) {
+    const newUser = new this.model({
+      firstName: req.body.firstName,
+      lastName: req.body.lastName,
+      email: req.body.email,
+      password: req.body.password,
```

Comment:
@hsedaqat1378 please don't use `password` in plain text. passwords always should be encrypted.

Reply:
also you should add validation rules for email to validate email format is in correct way